lib/matplotlib/axes/_axes.py

```diff
@@ -4382,6 +4384,18 @@ def do_patch(xs, ys, **kwargs):
                 flier_y = stats['fliers']
                 fliers.append(do_plot(flier_x, flier_y, **flier_kw))
 
+        # Set legend labels
+        if label:
+            box_or_med = boxes if showbox else medians
```

Comment:
As a remark: As long as we have the constraint to use only a single artist, the median is the best choice for `showbox=False`. The alternative would be to still use the box artist, which is a `Line2dD`, but it would appear as a single line not a box.